enhancements/installer/alibaba-cloud-ipi.md

```diff
@@ -0,0 +1,192 @@
+
+## Design Details
+
+![alt Architecture diagram](https://openshift-aliyun.oss-us-west-1.aliyuncs.com/enhancement/install_ipi.png?versionId=CAEQLRiBgICXh.6DyRciIDEyYTVlODg1ZDhhZDRlMjM4ZTgzMTYzMjc1NzdjOTU2)
+
+### Infrastructure Global Configuration
+
+#### API
+
+Alicloud will need to be added [Alicloud Environments in the API](https://github.com/openshift/api/blob/master/config/v1/types_infrastructure.go#L339-L355):
+
+#### Cluster Config Operator
+
+Cluster config operator will [need to be added](https://github.com/openshift/cluster-config-operator/blob/master/pkg/operator/kube_cloud_config/) and the new API vendored in.
+
+#### Installer Manifests & Cloud Provider Config
+
+The [Alicloud cloud-provider-config](https://github.com/openshift/installer/blob/master/pkg/asset/manifests/alicloud) will need to be added.
+
+#### Dev Preview -> Tech Preview
+
+#### Tech Preview -> GA
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+### Install Config
+
+New required fields will be added to `installconfig.platform.alicloud`.
+
+I'd expect fields for
+
+- Cluster
+  - imageid (required string): Image should be used to boot machines for cluster.
+  - regionid (required string): The region where the cluster will be created.
+  - VSwitchIds (optional array of strings): vswitches where cluster resources will be created. Leave unset to have the installer create vswitchs in a new VPC.
+  - userTags (optional object): the installer will add tags to all resources created.
+  - defaultMachinePlatform (optional object): Default machine pool properties.
+
+- Machine
+  - SystemDisk (optional object): Defines systemdisk for ECS instances.
+  - PerformanceLevel (optional String): [The performance level of systemdisk](https://www.alibabacloud.com/help/doc-detail/122389.htm). This is only valid for type essd.
+  - size (optional integer): Size of the system disk in gibibytes (GiB).
+  - Category (optional string): The type of disk.
+  - InstanceType (optional string): The ECS instance type.
+  - zones (optional array of strings): The availability zones used for machines.
+  - imageid (required string): The image should be used to boot machines.
+
+- Existing VPC & Vswitches:
+  - If users want to use some preexisting network resources instead of having the installer create one for them. Please use fields to point to these existing networking resources.
+    - VpcId (required String): A public VPC for the cluster.
+    - VSwitchIds (optional array of strings): vswitches where cluster resources will be created. The vswitches for between 1 and 3 availability zones.
+
+### Terraform
+
+Alicloud has [a Terraform provider](https://registry.terraform.io/providers/aliyun/alicloud/latest/docs).
+
+### Open Questions
+
+Suggestions for adopting the operators for the Alibaba Cloud.
+
+
+### Test Plan
+
+E2E and intergration tests should be developed for Alibaba Cloud.
+
+### Graduation Criteria
```

Comment:
We should have some items here. What are some minimum requirements we should meet before we consider the platform GA? At least I'd expect something like 1) E2E jobs covering cluster creation with `openshift-install` running periodically and stable; 2) IPI documentation available and published; 3) Test engineering validation by following the published documentation with successful results.

Reply:
The template offers some suggestions for minimum requirements for different support levels.